src/agreement/mod.rs

```diff
@@ -323,7 +303,7 @@ impl<NodeUid: Clone + Debug + Ord> Agreement<NodeUid> {
         self.epoch == 0 && self.estimated.is_none()
     }
 
-    fn handle_bval(&mut self, sender_id: &NodeUid, b: bool) -> Result<FaultLog<NodeUid>> {
+    fn handle_bval(&mut self, sender_id: &NodeUid, b: bool) -> Result<Step<NodeUid>> {
```

Comment:
Ideally we could distinguish beween the `Step` that the `DistAlgorithm` returns to the caller of a public function call and an intermediate data structure that is used to construct such a step.

Reply:
Possibly, yes. Or at least make some of the currently public `Step` methods `pub(crate)`.